modules/apps/ai-creator/ai-creator-openai-web/src/main/java/com/liferay/ai/creator/openai/web/internal/upload/AICreatorOpenAIUploadFileEntryHandler.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.ai.creator.openai.web.internal.upload;
+
+import com.liferay.document.library.kernel.model.DLFolderConstants;
+import com.liferay.document.library.kernel.service.DLAppService;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.repository.model.FileEntry;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.ServiceContextFactory;
+import com.liferay.portal.kernel.upload.UploadPortletRequest;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.MimeTypesUtil;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.upload.UploadFileEntryHandler;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+
+import java.net.URL;
+import java.net.URLConnection;
+
+import java.util.Date;
+import java.util.Iterator;
+import java.util.Set;
+
+import javax.portlet.PortletRequest;
+
+/**
+ * @author Roberto Díaz
+ */
+public class AICreatorOpenAIUploadFileEntryHandler
+	implements UploadFileEntryHandler {
+
+	public AICreatorOpenAIUploadFileEntryHandler(
+		DLAppService dlAppService, com.liferay.portal.kernel.util.File file) {
+
+		_dlAppService = dlAppService;
+		_file = file;
```

Comment:
(opt) FileImpl is not an OSGi service, so it is ok to call FileUtil directly here.